lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,6 +193,11 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
+        if user.is_anonymous():
```

Comment:
I still find it tightly coupled to have the `GradesBlockTransformer` need to make the stub-User _both_ `anonymous` and `known=False`.  Either we create a new User type for this Stub-User case, or expand the `known=False` case to include this Stub-User case.   By having Anonymous Users bypass code, we may inadvertently introduce security/other issues.